qemu/patches/qemu-register-ich6-ich9-soundhw.patch

```diff
@@ -0,0 +1,55 @@
+--- a/hw/audio/intel-hda.c	2019-12-12 21:20:47.000000000 +0300
++++ b/hw/audio/intel-hda.c	2020-05-28 22:01:45.919599285 +0300
+@@ -1303,9 +1303,9 @@ static const TypeInfo hda_codec_device_t
+ 
+ /*
+  * create intel hda controller with codec attached to it,
+- * so '-soundhw hda' works.
++ * so '-soundhw ich6' works.
+  */
+-static int intel_hda_and_codec_init(PCIBus *bus)
++static int intel_ich6_and_codec_init(PCIBus *bus)
+ {
+     DeviceState *controller;
+     BusState *hdabus;
+@@ -1318,6 +1318,19 @@ static int intel_hda_and_codec_init(PCIB
+     return 0;
+ }
+ 
++static int intel_ich9_and_codec_init(PCIBus *bus)
++{
++    DeviceState *controller;
++    BusState *hdabus;
++    DeviceState *codec;
++
++    controller = DEVICE(pci_create_simple(bus, -1, "ich9-intel-hda"));
++    hdabus = QLIST_FIRST(&controller->child_bus);
++    codec = qdev_create(hdabus, "hda-duplex");
++    qdev_init_nofail(codec);
++    return 0;
++}
++
+ static void intel_hda_register_types(void)
+ {
+     type_register_static(&hda_codec_bus_info);
+@@ -1325,7 +1338,8 @@ static void intel_hda_register_types(voi
+     type_register_static(&intel_hda_info_ich6);
+     type_register_static(&intel_hda_info_ich9);
+     type_register_static(&hda_codec_device_type_info);
+-    pci_register_soundhw("hda", "Intel HD Audio", intel_hda_and_codec_init);
```

Comment:
I'm confused by this patch. This rather looks like a libvirt bug than qemu issue. I think the proper fix should be extending libxl libvirt driver with similar function as qemu libvirt driver has here:
https://github.com/libvirt/libvirt/blob/6f60d2a8503ce8c624bce6b53bf7b68476f5056f/src/qemu/qemu_command.c#L4019-L4056

But, that's probably for the time when libxl will migrate to `-device` option for sound card (names seems to be different). For now, I think it's ok to patch it this way, but please do not break original device name (do not remove the line above).